core-bundle/src/Image/Preview/PreviewFactory.php

```diff
@@ -0,0 +1,197 @@
+            ->create(
+                $this->createPreviewFile($path, $this->getPreviewSizeFromImageSize($size)),
+                $size,
+                $options,
+            )
+        ;
+    }
+
+    /**
+     * @param int|string|array|PictureConfiguration|null $size
+     */
+    public function createPreviewPicture(string $path, $size = null, ResizeOptions $options = null): PictureInterface
+    {
+        // Unlike the Contao\Image\PictureFactory the PictureFactoryInterface
+        // does not know about ResizeOptions. We therefore check if the third
+        // argument of the 'create' method allows setting them.
+        $canHandleResizeOptions = static function (PictureFactoryInterface $factory): bool {
+            if ($factory instanceof PictureFactory) {
+                return true;
+            }
+
+            $createParameters = (new \ReflectionClass($factory))
+                ->getMethod('create')
+                ->getParameters()
+            ;
+
+            if (!isset($createParameters[2])) {
+                return false;
+            }
+
+            $type = $createParameters[2]->getType();
+
+            return $type instanceof \ReflectionNamedType && ResizeOptions::class === $type->getName();
+        };
+
+        $arguments = [$this->createPreviewFile($path, $this->getPreviewSizeFromImageSize($size)), $size];
+
+        if (null !== $options && $canHandleResizeOptions($this->pictureFactory)) {
+            $arguments[] = $options;
+        }
+
+        return $this->pictureFactory->create(...$arguments);
+    }
+
+    /**
+     * @param int|string|array|PictureConfiguration|null $size
+     */
+    public function createPreviewFigure(string $path, $size = null, ResizeOptions $options = null): Figure
+    {
+        return $this->createPreviewFigureBuilder($path, $size, $options)->build();
+    }
+
+    /**
+     * @param int|string|array|PictureConfiguration|null $size
+     */
+    public function createPreviewFigureBuilder(string $path, $size = null, ResizeOptions $options = null): FigureBuilder
+    {
+        return $this->imageStudio
+            ->createFigureBuilder()
+            ->fromPath($this->createPreviewFile($path, $this->getPreviewSizeFromImageSize($size)))
+            ->setSize($size)
+            ->setResizeOptions($options)
+        ;
+    }
+
+    /**
+     * @param int|string|array|PictureConfiguration|null $size
+     */
+    private function getPreviewSizeFromImageSize($size): int
+    {
+        // TODO: get correct size from size config
+        return 1024;
+    }
+
+    private function createCachePath(string $path, int $size): string
+    {
+        $hashData = [
+            Path::makeRelative($path, $this->cacheDir),
+            (string) $size,
+            (string) filemtime($path),
+        ];
+
+        $hash = substr(md5(implode('|', $hashData)), 0, 9);
+        $pathinfo = pathinfo($path);
+
+        return $hash[0].'/'.$pathinfo['filename'].'-'.substr($hash, 1);
```

Comment:
path util eye triggered :eye: 

```suggestion
        return Path::join($hash[0], $pathinfo['filename'].'-'.substr($hash, 1));
```